president.game.php

```diff
@@ -693,19 +697,44 @@ function stNewHand()
         $this->cards->moveAllCardsInLocation(null, "deck");
         $this->cards->shuffle('deck');
 
-        // Deal 13 cards to each players
-        // Create deck, shuffle it and give 13 initial cards
-        $i = 0;
-        while ($next_player_id != $activePlayer) {
-            $notifData = [];
-            $player_id = $next_player_id != '' ? $next_player_id : $activePlayer;
+        // Deal the cards, in pairs (unless there the number of cards in the deck is < 2 * playerCount)
+        $playerCount = $this->getPlayersNumber();
+        $deckSize = $this->cards->countCardInLocation("deck");
+        $deckLeft = $deckSize;
+
+        // The dealing player is the active player == the Beggar.
+        // He starts dealing at the player after him
+        $dealingPlayer = $activePlayer;
+        $firstPlayerToDealTo = self::getPlayerAfter( $dealingPlayer );
+        
+        while (true) {
+            $dealSize = ($deckLeft >= 2 * $playerCount) ? 2 : 1;
+            $iterPlayer = $firstPlayerToDealTo;
+            do {
+                if ($deckLeft == 0)
+                    break;
+                $dealtCards = $this->cards->pickCards( $dealSize, 'deck', $iterPlayer );
+                $deckLeft -= count($dealtCards);
+                $iterPlayer = self::getPlayerAfter( $iterPlayer );
+            }
+            while ($iterPlayer != $firstPlayerToDealTo);
+            if ($deckLeft == 0)
+                break;
+        }
+        // Notify players of their new Hand
+        {
```

Comment:
I'm not sure about this bracket ? maybe i'm missing something.

Reply:
Oh that's actually just a (maybe-not-so-nice) way of 'grouping a block of code together...
In a better IDE (I use VSCode) I would right-click - extract method on this and give it a name. I guess I was too lazy to actually do it. Will fix.